torch/csrc/DeviceSpec.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <Python.h>
+#include "torch/csrc/utils/device.h"
+
+struct THPDeviceSpec {
+  PyObject_HEAD
+  torch::DeviceType device_type;
+  int64_t device_index;
+  bool is_default;
```

Comment:
What are the semantics of this field? Can you please add a comment?

Reply:
it means 'default device for type', i.e. 'cuda' would have this set but 'cuda:1' wouldn't.

Reply:
Would `cuda:0` have it set?

Reply:
Yes.  To put it more explicitly, this controls whether `DeviceSpec.device_index` returns None or not.  So there is a distinction between `cuda` and `cuda:0` (because the default cuda device can change).

Reply:
Thanks for the explanation. Can you please write this down as a comment next to this field? It might be helpful for others in the future